boa3_test/examples/NEP17.py

```diff
@@ -0,0 +1,276 @@
+    # if the to_address is a smart contract, it must call the contracts onPayment
+    post_transfer(from_address, to_address, amount, data)
+    # and then it must return true
+    return True
+
+
+def post_transfer(from_address: UInt160, to_address: UInt160, amount: int, data: Any):
+    """
+    Checks if the one receiving NEP17 tokens is a smart contract and if it's one the onPayment method will be called
+
+    :param from_address: the address of the sender
+    :type from_address: UInt160
+    :param to_address: the address of the receiver
+    :type to_address: UInt160
+    :param amount: the amount of cryptocurrency that is being sent
+    :type amount: int
+    :param data: any pertinent data that might validate the transaction
+    :type data: Any
+    """
+    if to_address is not None:
+        contract = get_contract(to_address)
+        if contract is not None:
+            call_contract(to_address, 'onPayment', [from_address, amount, data])
+
+
+def mint(account: UInt160, amount: int):
+    """
+    Mints new tokens
+
+    :param account: the address of the account that is sending cryptocurrency to this contract
+    :type account: UInt160
+    :param amount: the amount of gas to be refunded
+    :type amount: int
+    :raise AssertionError: raised if amount is less than than 0
+    """
+    assert amount >= 0
+    if amount != 0:
+        current_total_supply = totalSupply()
+        account_balance = balanceOf(account)
+
+        put(SUPPLY_KEY, current_total_supply + amount)
+        put(account, account_balance + amount)
+
+        on_transfer(UInt160(), account, amount)     # TODO: Change to None when isinstance is implemented
+        post_transfer(UInt160(), account, amount, None)     # TODO: Change to None when isinstance is implemented
+
+
+@public
+def verify() -> bool:
+    """
+    When this contract address is included in the transaction signature,
+    this method will be triggered as a VerificationTrigger to verify that the signature is correct.
+    For example, this method needs to be called when withdrawing token from the contract.
+
+    :return: whether the transaction signature is correct
+    """
+    return check_witness(OWNER)
+
+
+@public
+def deploy() -> bool:
+    """
+    Initializes the storage when the smart contract is deployed.
+
+    :return: whether the deploy was successful. This method must return True only during the smart contract's deploy.
+    """
+    if not check_witness(OWNER):
+        return False
+
+    if get(SUPPLY_KEY).to_int() > 0:
+        return False
+
+    put(SUPPLY_KEY, TOKEN_TOTAL_SUPPLY)
+    put(OWNER, TOKEN_TOTAL_SUPPLY)
+
+    on_transfer(UInt160(), OWNER, TOKEN_TOTAL_SUPPLY)   # TODO: Change to None when isinstance is implemented
+    return True
+
+
+@public
+def onPayment(from_address: UInt160, amount: int, data: Any):
+    """
+    This method exists to check if this smart contract is receiving NEO or GAS so that it can mint a NEP17 token.
```

Comment:
Include the NEP17's definition in the comment and specify that this logic isn't a rule, just an example implementation
https://github.com/neo-project/proposals/blob/nep-17/nep-17.mediawiki#transfer